src/Solvers/plan_transforms.jl

```diff
@@ -174,45 +168,49 @@ function plan_transforms(grid::VerticallyStretched, storage, planner_flag)
+        # (Periodic, Flat) in the regular dimensions. So, we pretend like we need a 2D
+        # doubly-periodic transform (even if one dimension is Flat).
+        forward_periodic_plan = plan_forward_transform(storage, Periodic(), reg_dims, planner_flag)
+        backward_periodic_plan = plan_backward_transform(storage, Periodic(), reg_dims, planner_flag)
 
-        forward_transforms = tuple(DiscreteTransform(forward_periodic_plan, Forward(), grid, [1, 2]))
-        backward_transforms = tuple(DiscreteTransform(backward_periodic_plan, Backward(), grid, [1, 2]))
+        forward_transforms = tuple(DiscreteTransform(forward_periodic_plan, Forward(), grid, reg_dims))
+        backward_transforms = tuple(DiscreteTransform(backward_periodic_plan, Backward(), grid, reg_dims))
 
     else # we are on the GPU and we cannot / should not batch!
-        rs_storage = reshape(storage, (Ny, Nx, Nz))
+        grid_size = size(grid)
+        reshaped_storage1 = reshape(storage, (grid_size[reg_dims[1]], grid_size[reg_dims[2]], grid_size[irreg_dim]))
+        reshaped_storage2 = reshape(storage, (grid_size[reg_dims[2]], grid_size[reg_dims[1]], grid_size[irreg_dim]))
 
-        forward_plan_x = plan_forward_transform(storage,    topo[1](), [1], planner_flag)
-        forward_plan_y = plan_forward_transform(rs_storage, topo[2](), [1], planner_flag)
+        forward_plan_1 = plan_forward_transform(reshaped_storage1, topo[reg_dims[1]](), [1], planner_flag)
+        forward_plan_2 = plan_forward_transform(reshaped_storage2, topo[reg_dims[2]](), [1], planner_flag)
 
-        backward_plan_x = plan_backward_transform(storage,    topo[1](), [1], planner_flag)
-        backward_plan_y = plan_backward_transform(rs_storage, topo[2](), [1], planner_flag)
+        backward_plan_1 = plan_backward_transform(reshaped_storage1, topo[reg_dims[1]](), [1], planner_flag)
+        backward_plan_2 = plan_backward_transform(reshaped_storage2, topo[reg_dims[2]](), [1], planner_flag)
 
-        forward_plans = (forward_plan_x, forward_plan_y)
-        backward_plans = (backward_plan_x, backward_plan_y)
+        forward_plans  = Dict(reg_dims[1] => forward_plan_1,  reg_dims[2] => forward_plan_2)
+        backward_plans = Dict(reg_dims[1] => backward_plan_1, reg_dims[2] => backward_plan_2)
 
+        # Transform Flat topologies into Bounded
         unflattened_topo = Tuple(T() isa Flat ? Bounded : T for T in topo)
         f_order = forward_orders(unflattened_topo...)
         b_order = backward_orders(unflattened_topo...)
 
-        # Extract third dimension
-        f_order = Tuple(f_order[i] for i in findall(d -> d != 3, f_order))
-        b_order = Tuple(b_order[i] for i in findall(d -> d != 3, b_order))
+        # Extract irregularly-spaced dimension
+        f_order = Tuple(f_order[i] for i in findall(d -> d != irreg_dim, f_order))
+        b_order = Tuple(b_order[i] for i in findall(d -> d != irreg_dim, b_order))
```

Comment:
I think this clause is the only thing left for this PR to be ready for review. However I'm having a hard time getting this right since I'm not very familiar with the transforms and for some reason `Revise` doesn't work for me when I'm on the GPU nodes. I've asked on slack about it, but so far I haven't been able to fix it, which means my workflow for GPU development right now is super slow...

I'd appreciate some help/insight here if possible.

@simone-silvestri I believe you've worked on similar code recently, no? Any tips/suggestions?

Reply:
I am not sure you always have to reshape to put the irregular direction at the end. You probably just have to expose the transform direction